doc/nrf/ug_dev_model.rst

```diff
@@ -0,0 +1,514 @@
+timeframe to develop it and the need to update the |NCS| version used. All
+workflows are described under the following basic assumptions:
+
+- One or more applications are to be developed using the |NCS|
+- Additional board definitions might be required by the user
+- Additional libraries might be required by the user
+- The term "applicaiton" henceforth includes the application code and any board
+  definitions and libraries it requires
+- The application(s) will require updates of the |NCS| revision
+
+Worflow 1: Eschew Git and west
+------------------------------
+
+If you have your own version control tools, you might want to simply not use Git
+nor west at all, and instead rely on your own toolset. In that case, you will
+want to obtain a copy of the |NCS| on your filesystem and then manage the
+source code of both the SDK and your application yourself.
+
+Since no downloadable packages of the |NCS| are currently made available, the
+simples path to obtain the source code is to follow the instructions in the
+:ref:`corresponding section <dm-wf-get-ncs>` of the documentation. This will
+require you to install Git and west, but you can then ignore them from that
+point onwards. If you need to update the copy of the |NCS| you are working with
+you can :ref:`obtain the source code <dm-wf-get-ncs>` again or, if you have
+kept the original set of repositories,
+:ref:`update it instead <dm-wf-update-ncs>`.
+Once you have obtained a copy of the |NCS| source code, which will be
+self-contained in a single folder, you can then proceed to manage that code in
+any way you see fit.
+
+Worflow 2: Out of tree application repository
+---------------------------------------------
+
+A very simple approach to maintaining your application is to completely decouple
+it from the |NCS| repositories and instead host it wherever you prefer (be it
+Git, another version control system or simply on your hard drive). This is
+typically also known as "out of tree" application, meaning that the application,
+boards definitions and any other libraries are actually outside any of the
+repositories provided by the |NCS| and can in fact be placed anywhere at all.
+As long as you do not need to make any changes to any of the repositories of the
+|NCS|, you can use the procedures to :ref:`get the source code <dm-wf-get-ncs>`
+and later :ref:`update it <dm-wf-update-ncs>`, and manage your application
+separately, inside or outside the top folder of the |NCS|.
+
+If you choose to have your application outside the folder hierarchy of the
+|NCS|, the build system will find the location of the SDK through the
+:makevar:`ZEPHYR_BASE` environment variable, which is set either through a
+script or manually in an IDE. More information about application development and
+the |NCS| build and configuration system can be found in the :ref:`ncs-app-dev`
+documentation section.
+
+Worflow 3: App in a fork of `fw-nrfconnect-nrf`_
```

Comment:
For my money, this should be the recommended workflow, as it makes managing a small set of changes to other NCS projects (like zephyr) easier to keep in sync with upstream west.yml changes.

We could even guarantee that a top-level subdirectory glob of `nrf`, such as `nrf/user*`, will never be touched by nordic so customers know they'll never have rebase conflicts.

Reply:
I agree; this should be described first, in place of the "Eschew `Git` and `west`" workflow.

Reply:
I think I will leave the order as-is because:

1. The "no git, no west" is there just as a reminder that using the same tools we use is not mandatory, and I would like that to be clear
2. The "out of tree app" approach is by far the simplest, and encourages users not to touch the NCS repos and do everything cleanly out of tree. This will also increase the flow of patches towards upstream zephyr and/or ncs. Remember that even with an out of tree app you can always have local patches for temporary fixes, either manually or with a script.

Reply:
OK! Thanks for addressing the comment.